server/channels/app/user.go

```diff
@@ -1038,6 +1038,58 @@ func (a *App) userDeactivated(rctx request.CTX, userID string) *model.AppError {
 		rctx.Logger().Warn("unable to remove auth data by user id", mlog.Err(nErr))
 	}
 
+	if err = a.softDeleteUserDMChannels(rctx, userID, model.GetMillis()); err != nil {
+		rctx.Logger().Warn("Failed to archive DM channels for deactivated user", mlog.Err(err))
+	}
+
+	return nil
+}
+
+func (a *App) softDeleteUserDMChannels(rctx request.CTX, userID string, deleteTimestamp int64) *model.AppError {
```

Comment:
One potential issue here is, if the DM channel has more than 2 members (ie. a group DM channel), this will also automagically deleting that channel too. Just want to confirm this from the UX point of view to understand best approach. cc @matthewbirtch @abhijit-singh